vllm/model_executor/layers/quantization/utils/w8a8_utils.py

```diff
@@ -172,8 +173,13 @@ def apply(
         if use_per_token_if_dynamic is None:
             use_per_token_if_dynamic = self.use_per_token_if_dynamic
 
+        if out_dtype is None:
+            out_dtype = input.dtype
+
         # cutlass_scaled_mm supports per tensor/channel W and per tensor/token A
         if self.cutlass_fp8_supported:
+            assert input.dtype != current_platform.fp8_dtype(
+            ), "FP8 input to cutlass is not supported"
```

Comment:
Just want to make it clear this isn't an inherent limitation:
```suggestion
            assert input.dtype != current_platform.fp8_dtype(
            ), "FP8 input to cutlass is not currently supported"
```

Reply:
Sorry about the force push, accepting suggestions from the github UI breaks DCO...